### ISSUE TITLE
Error when validating a collection after removing an item

### ISSUE THREAD

**Issue by theosherman** [monterail/vuelidate#79]
This is similar or perhaps a duplicate to #73. But after reading that the issue was likely resolved, I tried using the latest repo and my issue is still occurring.

There is a issue with collection validating. I will try my best to explain how to reproduce this...
I am creating a very simple example form that involves a nested collection. It's a collection of contacts, and each contact has a collection of phone numbers nested inside. It's a list of lists.

The data...
```
form: {
  "contacts": [
    {
      "name": "John",
      "phoneNumbers": [
        {
          "phoneNumber": "(111) 111-1111",
          "type": "Home"
        }
      ]
    },
    {
      "name": "Luke",
      "phoneNumbers": [
        {
          "phoneNumber": "(222) 222-2222",
          "type": "Home"
        },
        {
          "phoneNumber": "(333) 333-3333",
          "type": "Work"
        }
      ]
    }
  ]
}
```

The validation logic...
```
validations: {
    form: {
      contacts: {
        $each: {
          name: { required, minLength: minLength(4) },
          phoneNumbers: {
            $each: {
              phoneNumber: { required }
            }
          }
        }
      }
    }
  }
```

The validation data created...
```
{
  "contacts": {
    "$each": {
      "0": {
        "name": {
          "required": true,
          "minLength": true,
          "$invalid": false,
          "$dirty": true,
          "$error": false,
          "$pending": false
        },
        "phoneNumbers": {
          "$each": {
            "0": {
              "phoneNumber": {
                "required": true,
                "$invalid": false,
                "$dirty": false,
                "$error": false,
                "$pending": false
              },
              "$invalid": false,
              "$dirty": false,
              "$error": false,
              "$pending": false
            },
            "$invalid": false,
            "$dirty": false,
            "$error": false,
            "$pending": false
          },
          "$invalid": false,
          "$dirty": false,
          "$error": false,
          "$pending": false
        },
        "$invalid": false,
        "$dirty": false,
        "$error": false,
        "$pending": false
      },
      "1": {
        "name": {
          "required": true,
          "minLength": true,
          "$invalid": false,
          "$dirty": true,
          "$error": false,
          "$pending": false
        },
        "phoneNumbers": {
          "$each": {
            "0": {
              "phoneNumber": {
                "required": true,
                "$invalid": false,
                "$dirty": false,
                "$error": false,
                "$pending": false
              },
              "$invalid": false,
              "$dirty": false,
              "$error": false,
              "$pending": false
            },
            "1": {
              "phoneNumber": {
                "required": true,
                "$invalid": false,
                "$dirty": false,
                "$error": false,
                "$pending": false
              },
              "$invalid": false,
              "$dirty": false,
              "$error": false,
              "$pending": false
            },
            "$invalid": false,
            "$dirty": false,
            "$error": false,
            "$pending": false
          },
          "$invalid": false,
          "$dirty": false,
          "$error": false,
          "$pending": false
        },
        "$invalid": false,
        "$dirty": false,
        "$error": false,
        "$pending": false
      },
      "$invalid": false,
      "$dirty": false,
      "$error": false,
      "$pending": false
    },
    "$invalid": false,
    "$dirty": false,
    "$error": false,
    "$pending": false
  },
  "$invalid": false,
  "$dirty": false,
  "$error": false,
  "$pending": false
}
```

The thing to note here is that contact 0 (John) has less phone numbers than than contact 1. This is necessary to produce the bug - having a contact that has less phone numbers than another contact that is further down in the array (higher index position).

So looking at the validation data, I'm going to summarize the created validation data in `$v` like this for easier visualization...
```
{
  0: {
    0: {...}
  },
  1: {
    0: {...},
    1: {...}
}
```
This is 2 items in contacts. Contact 0 has 1 phone number. Contact 1 has 2 phone numbers.

When I remove contact 0 from the array, I would expect the validation data to match the actual data. Which is for contact 1 to become the new contact 0, and the 2 phone number validations would come along for the ride (that would match the data). This is the data after removing contact 0...
```
form: {
  "contacts": [
    {
      "name": "Luke",
      "phoneNumbers": [
        {
          "phoneNumber": "(222) 222-2222",
          "type": "Home"
        },
        {
          "phoneNumber": "(333) 333-3333",
          "type": "Work"
        }
      ]
    }
  ]
}
```
This is my expectation of the validation data that would match it...
```
{
  0: {
    0: {...},
    1: {...}
}
```

What really happens is this...
```
{
  0: {
    0: {...}
}
```

AAHH! You see? No more second phone number. It appears that the contact 1 validation becomes 0 as expected, however, the phone number validations nested inside contact 1 missed the bus.

What will ultimately happen as a result of this is an error (`phoneNumber` is undefined) with my validation class styling...
```
.well(v-for="(contact, i) in form.contacts")
  ...
  .row(v-for="(phoneNumber, j) in contact.phoneNumbers")
    ..
    .form-group(:class="{ 'has-error': $v.form.contacts.$each[i].phoneNumbers.$each[j].phoneNumber.$error }")
```
Because the data that v-for is iterating over has that second phone number, and the validation data doesn't have that second phone number.

Hope I explained this alright. It's not a strength of mine. Let me know if you have questions.

Source is at...
https://github.com/theosherman/vuejs-demo/blob/master/src/components/Form/AdvancedForm.vue

Thank you for your time. I love your work!

**Comment by Frizi** [comment]
I was able to tweak your example repository and actually reproduce this bug, but it's a bit too complicated to comfortably work with. I'll have to create a minimal example and write a test for it. It's definitely a vuelidate bug though. thanks for the report!

**Comment by Frizi** [comment]
Reproduction:
https://jsfiddle.net/vaxnqoue/1/

Click on first item, flag on the new first will turn into "false".

**Comment by Frizi** [comment]
Fixed in v0.4.0
